### PR TITLE
Update of AOD reader / writer

### PR DESCRIPTION
**Pull request by pbuehler**
. this is to work with files containing folders TF_%d

### CODE REVIEW

File: Framework/Core/include/Framework/runDataProcessing.h [hunk 1]
```diff
@@ -128,6 +128,13 @@ int main(int argc, char** argv)
     workflowOptions.push_back(ConfigParamSpec{"pipeline", VariantType::String, "", {"override default pipeline size"}});
     workflowOptions.push_back(ConfigParamSpec{"dangling-outputs-policy", VariantType::String, "file", {"what to do with dangling outputs. file: write to file, fairmq: send to output proxy"}});
 
+    // options for AOD writer
+    workflowOptions.push_back(ConfigParamSpec{"aod-writer-json", VariantType::String, {"Name of the json configuration file"}});
+    workflowOptions.push_back(ConfigParamSpec{"res-file", VariantType::String, {"Default name of the output file"}});
```
Comment (jgrosseo) [inline Framework/Core/include/Framework/runDataProcessing.h]: We could also prepend aod-writer to the other 3 options

Comment (ktf) [reply Framework/Core/include/Framework/runDataProcessing.h]: i agree.

Comment (pbuehler) [reply Framework/Core/include/Framework/runDataProcessing.h]: and then also aod-reader-...for all the reader options?

Comment (jgrosseo) [reply Framework/Core/include/Framework/runDataProcessing.h]: We only have one reader, so I would not touch that for now.

**Comment by jgrosseo** [comment]
Most tests seem fine now, except test_Framework_test_WorkflowHelpers
Is this ready to be tested locally by us?

**Comment by pbuehler** [comment]
Wait a sec, I am updating the PR ...

**Comment by pbuehler** [comment]
@jgrosseo it is ready to be tested locally.

**Comment by jgrosseo** [comment]
I am trying to read an alien file where the path is in text file alien.txt, and I get

[23034:internal-dpl-aod-reader]: Info in <TJAlienFile::Open>: Accessing file /alice/data/2015/LHC15o/000244918/pass5_lowIR/PWGZZ/Run3_Conversion/96_20201013-1346_child_1/0028/AO2D.root in SE <ALICE::CERN::EOS>
[23034:internal-dpl-aod-reader]: [METRIC] matcher_variables/236,1 null 1603281460226 hostname=pcalice57,datapError in <TFile::Open>: no url specified
[23034:internal-dpl-aod-reader]: rocessor_id=internal-dpl-aod-reader
[23034:internal-dpl-aod-reader]: [13:57:40][ERROR] Exception caught: Couldn't open file ""!
[23034:internal-dpl-aod-reader]: [13:57:40][STATE] Exiting FairMQ state machine
[23034:internal-dpl-aod-reader]: [13:57:40][ERROR] Unhandled o2::framework::runtime_error reached the top of main, device shutting down. Details follow:

Does this work for you?

**Comment by pbuehler** [comment]
yes, e.g.
o2-analysistutorial-aodwriter --readers 3 --aod-file @alienfiles.txt --aod-writer-keep dangling

.
.
[10734:internal-dpl-aod-reader]: Info in <TJAlienFile::Open>: Accessing file /alice/data/2015/LHC15o/000244918/pass5_lowIR/PWGZZ/Run3_Conversion/96_20201013-1346_child_1/0028/AO2D.root in SE <ALICE::GSI::SE2>
[10733:internal-dpl-aod-reader]: Info in <TJAlienFile::Open>: Accessing file /alice/data/2015/LHC15o/000244918/pass5_lowIR/PWGZZ/Run3_Conversion/96_20201013-1346_child_1/0012/AO2D.root in SE <ALICE::GSI::SE2>
[10732:internal-dpl-aod-reader]: Info in <TJAlienFile::Open>: Accessing file /alice/data/2015/LHC15o/000244918/pass5_lowIR/PWGZZ/Run3_Conversion/96_20201013-1346_child_1/0009/AO2D.root in SE <ALICE::GSI::SE2>
.
.
and processes successfully.

**Comment by jgrosseo** [comment]
It works also with 1 reader and just 1 file in the txt?

**Comment by jgrosseo** [comment]
I tried the histogram example

**Comment by jgrosseo** [comment]
I think the problem is if the text file has an empty line, those should be skipped by default

**Comment by pbuehler** [comment]
o2-analysistutorial-histograms --aod-file @alienfiles.txt

with 1 file only works as well.

**Comment by pbuehler** [comment]
> I think the problem is if the text file has an empty line, those should be skipped by default

indeed empty lines cause problems. Will change that.

**Comment by jgrosseo** [comment]
Thanks for fixing this problem.

I see another problem when writing output. The following workflow works fine (but has nothing to write):
` o2-analysis-event-selection --aod-file AO2D.root | o2-analysis-trackextension | o2-analysis-trackselection | o2-analysis-multiplicity-table | o2-analysis-centrality-table | o2-analysis-timestamp | o2-analysis-run2-matcher`

If I add the writer task (to actually get my tables), it fails
`o2-analysis-event-selection --aod-file AO2D.root --resources-monitoring 1 --aod-writer-keep AOD/CFCOLLISION/0,AOD/CFTRACK/0 --aod-writer-res-file tables | o2-analysis-trackextension | o2-analysis-trackselection | o2-analysis-multiplicity-table | o2-analysis-centrality-table | o2-analysis-timestamp | o2-analysis-run2-matcher | o2-analysis-filter-cf`
with
`[27476:internal-dpl-aod-reader]: [16:33:42][ERROR] Exception caught: Couldn't get TTree "TF_0/O2evsel" from "aod.root"`
This is strange because this table is produced by o2-analysis-event-selection which is part of the workflow.

So somehow the topology is not right.

**Comment by pbuehler** [comment]
yes I see similar things. 
The message - Couldn't get TTree "TF_0/O2evsel" from "aod.root" - also tells that not AO2D.root is used, as specified on the command line, but the default file. This by the way only happens in "complex" pipelines. Not sure how the workflow gets messed up.

**Comment by pbuehler** [comment]
@jgrosseo since the writer options are now global, provide them to all tasks. That seems to create a proper WorkFlow but still no output is written.

**Comment by jgrosseo** [comment]
Thanks. This indeed solves it.
@eulisse where does this problem come from that if you provide a global option not to all tasks you get undefined behaviour?

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
For me everything looks good now. Can be merged once the tests pass.

**Pull request merged**